Detectors/ITSMFT/common/reconstruction/src/DigitPixelReader.cxx

```diff
@@ -173,10 +180,10 @@ bool DigitPixelReader::getNextChipData(ChipPixelData& chipData)
 
     // loop over chip pixels
     for (int iPixel{0}, iDigitNext{0}; iPixel < initialSize; ++iPixel) {
-      auto& pixel = chipData.getData()[iPixel];
+      const o2::itsmft::PixelData pixel = chipData.getData()[iPixel]; // we need a copy to avoid the reference to be invalidated
```

Comment:
@shahor02 this is what caused the undefined behaviour: the possible reallocation of the `chipData.getData()` vector upon an emplace_back() caused to corrupt the data pointed.